webknossos-datastore/app/com/scalableminds/webknossos/datastore/helpers/DatasetDeleter.scala

```diff
@@ -31,23 +39,244 @@ trait DatasetDeleter extends LazyLogging with DirectoryConstants {
+    for {
+      dataSourceId <- Fox.successful(DataSourceId(datasetName, organizationId))
+      layersAndLinkedMagsOpt <- Fox.runOptional(remoteWKClient)(_.fetchPaths(dataSourceId))
+      exceptionBoxes = layersAndLinkedMagsOpt match {
+        case Some(layersAndLinkedMags) =>
+          layersAndLinkedMags.map(layerMagLinkInfo =>
+            handleLayerSymlinks(dataSourceId, layerMagLinkInfo.layerName, layerMagLinkInfo.magLinkInfos.toList))
+        case None => Seq(tryo {})
+      }
+      _ <- Fox.combined(exceptionBoxes.toList.map(Fox.box2Fox)) ?~> "Failed to move symlinks"
+      affectedDataSources = layersAndLinkedMagsOpt
+        .map(layersAndMags => layersAndMags.map(_.magLinkInfos.map(m => m.linkedMags.map(_.dataSourceId))))
+        .getOrElse(List())
+        .flatten
+        .flatten
+      _ <- updateDatasourceProperties(affectedDataSources)
+    } yield ()
+
+  private def getFullyLinkedLayers(linkedMags: List[MagLinkInfo]): Option[Seq[(DataSourceId, String)]] = {
+    val allMagsLocal = linkedMags.forall(_.mag.hasLocalData)
+    val allLinkedDatasetLayers = linkedMags.map(_.linkedMags.map(lm => (lm.dataSourceId, lm.dataLayerName)))
+    // Get combinations of datasourceId, layerName that link to EVERY mag
+    val linkedToByAllMags = allLinkedDatasetLayers.reduce((a, b) => a.intersect(b))
+    if (allMagsLocal && linkedToByAllMags.nonEmpty) {
+      Some(linkedToByAllMags)
+    } else {
+      None
+    }
+  }
+
+  private def relativizeSymlinkPath(targetPath: Path, originPath: Path): Path = {
+    val absoluteTargetPath = targetPath.toAbsolutePath
+    val relativeTargetPath = originPath.getParent.toAbsolutePath.relativize(absoluteTargetPath)
+    relativeTargetPath
+  }
+
+  private def getMagPath(basePath: Path, magInfo: DataSourceMagInfo): Path =
+    basePath
+      .resolve(magInfo.dataSourceId.organizationId)
+      .resolve(magInfo.dataSourceId.directoryName)
+      .resolve(magInfo.dataLayerName)
+      .resolve(magInfo.mag.toMagLiteral(true))
+
+  private def updateDatasourceProperties(dataSourceIds: List[DataSourceId])(
+      implicit ec: ExecutionContext): Fox[List[Unit]] =
+    // We need to update locally explored datasets, since they now may have symlinks where previously they only had the
+    // path property set.
+    Fox.serialCombined(dataSourceIds)(dataSourceId => {
+      val propertiesPath = dataBaseDir
+        .resolve(dataSourceId.organizationId)
+        .resolve(dataSourceId.directoryName)
+        .resolve(GenericDataSource.FILENAME_DATASOURCE_PROPERTIES_JSON)
+      if (Files.exists(propertiesPath)) {
+        JsonHelper.validatedJsonFromFile[DataSource](propertiesPath, dataBaseDir) match {
+          case Full(dataSource) =>
+            val updatedDataSource = dataSource.copy(dataLayers = dataSource.dataLayers.map {
+              case dl: DataLayerWithMagLocators =>
+                if (dl.mags.forall(_.path.exists(_.startsWith("file://")))) {
```

Comment:
```suggestion
                if (dl.mags.forall(_.path.exists(_.startsWith(s"${DataVaultService.schemeFile}://")))) {
```

That way this will be listed as a usage of this file scheme.